tartufo/scanner.py

```diff
@@ -351,18 +357,23 @@ def rule_matches(rule: Rule, string: str, line: str, path: str) -> bool:
 
         :param rule: Rule to perform match
         :param string: string to match against rule pattern
+        :param line: Source line containing string of interest
         :param path: path to match against rule path_pattern
         :return: True if string and path matched, False otherwise.
         """
         match = False
-        if rule.re_match_type == "match":
+        if rule.re_match_scope == Scope.Word:
+            scope = string
+        else:
```

Comment:
We should probably explicitly match `Scope.Line` here and error out for invalid values.